processing/src/main/java/org/apache/druid/segment/filter/OrFilter.java

```diff
@@ -107,6 +273,27 @@ public <T> T computeBitmapResult(BitmapResultFactory<T> bitmapResultFactory, boo
             () -> bitmapColumnIndices.stream().map(x -> x.computeBitmapResult(bitmapResultFactory, includeUnknown)).iterator()
         );
       }
+
+      @Nullable
+      @Override
+      public <T> T computeBitmapResult(
+          BitmapResultFactory<T> bitmapResultFactory,
+          int selectionRowCount,
+          int totalRowCount,
+          boolean includeUnknown
+      )
+      {
+        List<T> results = Lists.newArrayListWithCapacity(bitmapColumnIndices.size());
+        for (BitmapColumnIndex index : bitmapColumnIndices) {
+          final T r = index.computeBitmapResult(bitmapResultFactory, selectionRowCount, totalRowCount, includeUnknown);
+          if (r == null) {
+            // all or nothing
+            return null;
+          }
```

Comment:
the java doc for `BitmapColumnIndex.computeBitmapResult()` doesn't include a definition of what it means to return `null`.  If a bitmap index cannot be computed, I would've expected `index` itself to be null, so either that is never `null` OR `null` means that it's an empty bitmap.  If `null` means the empty bitmap, then this if statement is wrong (you just skip the clause).  If it is never null, then the if statement is unnecessary and potentially causes confusion.

Reply:
The 2 argument version is not expected to be null, however the 4 argument version may return null and the javadocs reflect that https://github.com/apache/druid/pull/15838/files#diff-0524c5910c725c7074f7383e79ced3937827cdaae4170916f7f2ff017b21669dR56, which is how this new stuff can work.

If you remember, building the `BitmapColumnIndex` is not doing the actual work, rather it returns a thing that you call the 'compute' methods on to do the actual work to build the bitmaps. If `BitmapColumnIndex` is null then we cannot possibly use an index, while if not null, the 2 argument version will _always_ build the index. The 4 argument version is what allows this new functionality to let the `BitmapColumnIndex` decide if it would have to do too much work to compute the index based on the selected row count so far.

This is a lot cleaner with the way things currently work, otherwise the all of the 'semantic' indexes classes returned by the `.as` of the index supplier (`ArrayElementIndexes`, `DruidPredicateIndexes`, `LexicographicalRangeIndexes`, `NullValueIndex`, `NumericRangeIndexes`, `StringValueSetIndexes`, `Utf8ValueSetIndexes`, `ValueIndexes`) would all need to have their methods that build `BitmapColumnIndex` updated to also take the selected and total row count parameters when building the `BitmapColumnIndex`. This was a much bigger surface area than just updating `BitmapColumnIndex` itself, since it is the common interface that _all_ of these implementations spit out. I'm very much in favor of doing it this way because it is both significantly cleaner and a lot simpler to implement.

Reply:
Do we still need the 2-argument method?  :)

Reply:
well, the search query is still calling the 2 argument version and computing the indexes directly from the filter instead of going through the bundle (also a bunch of benchmarks are calling it), so for now I was thinking about leaving it for this PR at least. I'll need to switch search query over to using the bundles anyway though, since its the real replacement for the 'auto' strategy that we removed, so I can look into removing the 2 argument version at that point.